### PR TITLE
Migrate ner-tagger

### PR DESCRIPTION
**Pull request by AlasDiablo**

### CODE REVIEW

File: services/ner-tagger/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# syntax=docker/dockerfile:1.2
+
+####
+# Get all web services dvc files
```
Comment (parmentf) [inline services/ner-tagger/Dockerfile:4]: **typo:** ‏
```suggestion
# Get all web service's dvc files
```

File: services/ner-tagger/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    git init && \
+    dvc init && \
+    dvc remote add -d webdav-remote "$(cat /run/secrets/webdav_url)" && \
+    dvc remote modify --local webdav-remote user "$(cat /run/secrets/webdav_login)" && \
+    dvc remote modify --local webdav-remote password "$(cat /run/secrets/webdav_password)"
+COPY ./v1/v1.dvc /dvc
+RUN dvc pull -v
+
+####
+# Build and install web services dependcies en files
```
Comment (parmentf) [inline services/ner-tagger/Dockerfile:22]: **typo:** ‏
```suggestion
# Build and install web services dependencies files
```

File: services/ner-tagger/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# ws-ner-tagger@0.0.0
```
Comment (parmentf) [inline services/ner-tagger/README.md]: **todo:** ‏Let's use 1.0.6 as a base
```suggestion
# ws-ner-tagger@1.0.6
```

File: services/ner-tagger/README.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+
+Détéction d'entités nommées dans des textes anglais
```
Comment (parmentf) [inline services/ner-tagger/README.md:3]: **suggestion:** ‏That's the Persée BAIP tagger
```suggestion
## Persée BAIP Tagger

Détection d'entités nommées dans des textes anglais
```

File: services/ner-tagger/package.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
+{
+  "private": true,
+  "name": "ws-ner-tagger",
+  "version": "0.0.0",
```
Comment (parmentf) [inline services/ner-tagger/package.json]:
```suggestion
  "version": "1.0.6",
```

File: services/ner-tagger/package.json [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  "description": "Détéction d'entités nommées dans des textes anglais",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/Inist-CNRS/web-services.git"
+  },
+  "keywords": [
+    "ezmaster"
+  ],
+  "author": " <>",
```
Comment (parmentf) [inline services/ner-tagger/package.json]:
```suggestion
  "author": " <Lucas Anki>",
```

File: services/ner-tagger/swagger.json [hunk 7]
```diff
@@ -0,0 +1,33 @@
+{
+    "openapi": "3.0.0",
+    "info": {
+        "title": "ner-tagger - Détéction d'entités nommées dans des textes anglais",
+        "description": "Détecter des entités géographiques dans des textes anglais et d'organismes scolaires, de localisation dans des bulletins administratifs de l'instruction publique",
+        "version": "0.0.0",
```
Comment (parmentf) [inline services/ner-tagger/swagger.json]:
```suggestion
        "version": "1.0.6",
```

File: services/ner-tagger/swagger.json [hunk 8]
```diff
@@ -0,0 +1,33 @@
+{
+    "openapi": "3.0.0",
+    "info": {
+        "title": "ner-tagger - Détéction d'entités nommées dans des textes anglais",
```
Comment (parmentf) [inline services/ner-tagger/swagger.json:4]: **typo:** ‏
```suggestion
        "title": "ner-tagger - Détection d'entités nommées dans des textes anglais",
```

File: services/ner-tagger/v1/geoTagger/geoTagger.ini [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+post.parameters.0.in = query
+post.parameters.0.name = path
+post.parameters.0.schema.type = string
+post.parameters.0.description = The path in each object to enrich with an Python script
+post.parameters.1.in = query
+post.parameters.1.name = indent
+post.parameters.1.schema.type = boolean
+post.parameters.1.description = Indent or not the JSON Result
+
+[use]
```
Comment (parmentf) [inline services/ner-tagger/v1/geoTagger/geoTagger.ini:19]: **suggestion:** ‏Generate example metadata, and insert them here

See [generate:example-metadata](https://github.com/Inist-CNRS/web-services/blob/main/SCRIPTS.md#generateexample-metadata)

File: services/ner-tagger/v1/perseeBaipTagger/tagger.ini [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+post.parameters.0.in = query
+post.parameters.0.name = path
+post.parameters.0.schema.type = string
+post.parameters.0.description = The path in each object to enrich with an Python script
+post.parameters.1.in = query
+post.parameters.1.name = indent
+post.parameters.1.schema.type = boolean
+post.parameters.1.description = Indent or not the JSON Result
+
+[use]
```
Comment (parmentf) [inline services/ner-tagger/v1/perseeBaipTagger/tagger.ini:19]: **suggestion:** ‏Generate example metadata, and insert them here

See [generate:example-metadata](https://github.com/Inist-CNRS/web-services/blob/main/SCRIPTS.md#generateexample-metadata)

File: services/ner-tagger/v1/perseeBaipTagger/tagger.ini [hunk 11]
```diff
@@ -0,0 +1,36 @@
+# OpenAPI Documentation - JSON format (dot notation)
+mimeType = application/json
+
+post.description = Détecte les organismes scolaires et localisations dans les baip
```
Comment (parmentf) [inline services/ner-tagger/v1/perseeBaipTagger/tagger.ini:4]: **typo:** ‏
```suggestion
post.description = Détecte les organismes scolaires et localisations dans les BAIP
```

**Review by parmentf — CHANGES_REQUESTED** [review]

File: services/ner-tagger/package.json [hunk 12]
```diff
@@ -0,0 +1,42 @@
+{
+  "private": true,
+  "name": "ws-ner-tagger",
+  "version": "1.0.6",
+  "description": "Détéction d'entités nommées dans des textes anglais",
```
Comment (parmentf) [inline services/ner-tagger/package.json:5]: **typo:** ‏
```suggestion
  "description": "Détection d'entités nommées dans des textes anglais",
```

**Review by parmentf — CHANGES_REQUESTED** [review]

File: services/ner-tagger/README.md [hunk 13]
```diff
@@ -0,0 +1,7 @@
+# ws-ner-tagger@1.0.6
+
+## Persée BAIP Tagger
+
+Détection d'entités nommées dans des textes anglais
+
+Détecter des entités géographiques dans des textes anglais et d'organismes scolaires, de localisation dans des bulletins administratifs de l'instruction publique
```
Comment (parmentf) [inline services/ner-tagger/README.md:7]: **suggestion:** Enlever le titre de section `Persée BAIP Tagger`
‏
```suggestion
Détection d'entitées nommées:

- entités géographiques dans des textes anglais
- organismes scolaires, de localisation dans des bulletins administratifs de l'instruction publique (BAIP)
```

File: services/ner-tagger/config.json [hunk 14]
```diff
@@ -0,0 +1,14 @@
+{
+    "environnement": {
+        "EZS_TITLE": "Détéction d'entités nommées dans des textes anglais",
```
Comment (parmentf) [inline services/ner-tagger/config.json:3]: **suggestion:** ‏Not only in English texts
```suggestion
        "EZS_TITLE": "Détection d'entités nommées dans des textes",
```

File: services/ner-tagger/package.json [hunk 15]
```diff
@@ -0,0 +1,42 @@
+{
+  "private": true,
+  "name": "ws-ner-tagger",
+  "version": "1.0.6",
+  "description": "Détection d'entités nommées dans des textes anglais",
```
Comment (parmentf) [inline services/ner-tagger/package.json:5]:
```suggestion
  "description": "Détection d'entités nommées dans des textes",
```

File: services/ner-tagger/swagger.json [hunk 16]
```diff
@@ -0,0 +1,33 @@
+{
+    "openapi": "3.0.0",
+    "info": {
+        "title": "ner-tagger - Détection d'entités nommées dans des textes anglais",
```
Comment (parmentf) [inline services/ner-tagger/swagger.json:4]:
```suggestion
        "title": "ner-tagger - Détection d'entités nommées dans des textes",
```

File: services/ner-tagger/swagger.json [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            "url": "http://vptdmservices.intra.inist.fr:49225/",
+            "description": "Latest version for production",
+            "#DISABLED#x-profil": "Standard"
+        }
+    ],
+    "tags": [
+        {
+            "name": "ner-tagger",
+            "description": "Détéction d'entités nommées dans des textes anglais",
```
Comment (parmentf) [inline services/ner-tagger/swagger.json:26]:
```suggestion
            "description": "Détéction d'entités nommées dans des textes",
```

File: services/ner-tagger/v1/geoTagger/geoTagger.ini [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+post.responses.default.content.application/json.example.0.value.3 = France
+post.responses.default.content.application/json.example.0.value.4 = Europe
+post.responses.default.content.application/json.example.1.id = 2
+post.responses.default.content.application/json.example.1.value.0 = Wuhan
+post.responses.default.content.application/json.example.1.value.1 = China
+
+[use]
+plugin = @ezs/spawn
+plugin = @ezs/basics
+plugin = @ezs/analytics
```
Comment (parmentf) [inline services/ner-tagger/v1/geoTagger/geoTagger.ini:36]: **quibble (non-blocking):** ‏`analytics` is no more required
```suggestion
```

File: services/ner-tagger/v1/perseeBaipTagger/tagger.ini [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+post.responses.default.content.application/json.example.1.value.basicOrg.0 = ecole
+post.responses.default.content.application/json.example.1.value.basicOrg.1 = lycée
+post.responses.default.content.application/json.example.1.value.basicOrg.2 = école
+post.responses.default.content.application/json.example.1.value.basicOrg.3 = école
+post.responses.default.content.application/json.example.1.value.operation = []
+
+[use]
+plugin = @ezs/spawn
+plugin = @ezs/basics
+plugin = @ezs/analytics
```
Comment (parmentf) [inline services/ner-tagger/v1/perseeBaipTagger/tagger.ini:52]: **chore:** ‏`analytics` is no more required
```suggestion
```

File: bases/ezs-python-pytorch-server/Dockerfile [hunk 20]
```diff
@@ -51,14 +51,20 @@ RUN apt update && apt install -y build-essential make gcc && \
...
 # Create a cache directory
 RUN mkdir /app/.cache
 
 # Convert every file owner
 RUN chown daemon:daemon -R /app
 
 # Set the transformers cache directory
-ENV TRANSFORMERS_CACHE=/app/.cache/transformers
+# ENV TRANSFORMERS_CACHE=/app/.cache/transformers # Old transformers cache directory environement variables
+ENV HF_HOME=/app/.cache/transformers
```
[the remark this answers is not in the document]
Comment (AlasDiablo) [reply bases/ezs-python-pytorch-server/Dockerfile:67]: Don't no, but it seem to work like that...

File: bases/ezs-python-pytorch-server/Dockerfile [hunk 21]
```diff
@@ -51,14 +51,20 @@ RUN apt update && apt install -y build-essential make gcc && \
     apt remove -y --purge build-essential make gcc && \
     apt autoremove -y
 
+# Make sure that .npm folder is readable by the daemon user
```
[the remark this answers is not in the document]
Comment (AlasDiablo) [reply bases/ezs-python-pytorch-server/Dockerfile:54]: Yes, is a copy/past typo

File: services/ner-tagger/swagger.json [hunk 22]
```diff
@@ -23,7 +23,7 @@
     "tags": [
         {
             "name": "ner-tagger",
-            "description": "Détéction d'entités nommées dans des textes anglais",
+            "description": "Détéction d'entités nommées dans des textes",
```
Comment (parmentf) [inline services/ner-tagger/swagger.json:26]: **typo:** ‏
```suggestion
            "description": "Détection d'entités nommées dans des textes",
```

**Review by parmentf — APPROVED** [review]
There is a little change to do, but it's not critical.

**Pull request merged**